src/metldata/schemapack_/builtin_transformations/infer_relations/path/path.py

```diff
@@ -0,0 +1,119 @@
+    attribute) and an element-based ("elements" attribute) one.
+
+    In the string-based representation ("path_string" attribute), the first and the last
+    word correspond the name of the source and target class, respectively. ">" and "<"
+    means indicate active (left class relations the right one) and passive (the left
+    class is referenced by the right one). Parentheses attached to these angles thereby
+    indicate the property name of the referencing class. E.g. "class_a(class_b)>class_b"
+    means that the source class "class_a" has a property "class_b" that relations the
+    target class "class_b". Or "class_a<(class_a)class_b" means that the source
+    class "class_a" is relation by the target class "class_b" via its properties
+    "class_a". Reference paths can also involve additional classes. E.g. a string of
+    "class_a<(class_a)class_b(class_c)>class_c" means that
+    a relation from the source class "class_a" to the target class "class_c" can be
+    established via an additional class "class_b". Any inserted spaces or newlines will
+    be ignored. So the following paths are equivalent:
+        - "class_a (class_b)> class_b"
+        - "class_a
+            (class_b)>
+            class_b"
+
+    A relation path consists of one or more elements. An element is a relationship
+    between two classes. Reference paths that establish a direct relationship between
+    source and target classes without the use of additional classes have only one
+    element (e.g. in string representations "class_a(class_b)>class_b" or
+    "class_a<(class_a)class_b"). More complex paths consist of multiple elements.
+    E.g. the path "class_a<(class_a)class_b(class_c)>class_c" can be decomposed
+    into the elements: "class_a<(class_a)class_b" and
+    "class_b(class_c)>class_c".
+
+    The elements of a ReferencePath are stored in the "elements" attribute as a list
+    of ReferencePathElement objects that are optimized for programmatic use.
+
+    The "source" attribute provides the source class of the path while the
+    "target" attribute provides the target class of the path.
+    """
+
+    def __init__(self, *, path_str: str):
+        """Construct relation path from a string-based representation."""
+        self.path_str = clean_path_str(path_str=path_str)
+        self.elements = path_str_to_object_elements(path_str=self.path_str)
+        self.source = self.elements[0].source
+        self.target = self.elements[-1].target
+
+    @classmethod
+    def validate(cls, value, info: ValidationInfo) -> "RelationPath":
+        """A validator for pydantic."""
+        if isinstance(value, cls):
+            return value
+
+        if not isinstance(value, str):
+            raise ValueError("A string is required.")
+
+        try:
+            return cls(path_str=value)
+        except ValidationError as error:
+            raise ValueError(str(error)) from ValidationError
+
+    @classmethod
+    def __get_validators__(cls):
```

Comment:
I think this is deprecated in Pydantic v2.

Reply:
I already have that ob my refactoring list for after the POC.